central/reports/scheduler/schedule.go

```diff
@@ -375,13 +406,22 @@ func (s *scheduler) getReportData(ctx context.Context, rQuery *common.ReportQuer
 	return r, nil
 }
 
-func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery string) (common.Result, error) {
-	offset := 0
+func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery string, deploymentsQuery *v1.Query) (common.Result, error) {
+	offset := paginatedQueryStartOffset
 	var resultData common.Result
 	for {
 		var gqlQuery string
+		gqlPaginationOffset := offset
 		if env.PostgresDatastoreEnabled.BooleanSetting() {
 			gqlQuery = reportQueryPostgres
+			if features.ObjectCollections.Enabled() {
+				deploymentIds, err := s.getDeploymentIDs(ctx, deploymentsQuery, int32(offset))
```

Comment:
Since this is not happening in one transaction, it is likely that the ID fetched on the previous page appears in the next page because of the changing DB state. You have to make sure seen ids are not fetched again.

I do not see a point of paginating the IDs if the report itself is not paginated. graphQL is paginated because it pulls out full objects.

Reply:
Ah great catch, didn't think of that case. Changed it. Fetching all deploymentIDs in a single transaction now and iterating over the ids in a paginated manner to exec graphQL query.